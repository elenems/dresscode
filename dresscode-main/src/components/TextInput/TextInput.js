import React from "react";
import PropTypes from "prop-types";
function TextInput(props) {
  const { name, place, value, handleChange, type, error, required } = props;
  const inputStyle = error
    ? { borderColor: "#d0021b" }
    : { borderColor: "#b1b1b1" };
  return (
    <div className="input-container">
      {required === false ? (
        <label className="required" htmlFor={name}>
          {name}
        </label>
      ) : (
        <label htmlFor={name}>{name}</label>
      )}
      <input
        style={inputStyle}
        onChange={handleChange}
        id={name}
        type={type}
        value={value}
        placeholder={place}
      />
      {error ? <span className="error">{error}</span> : null}
    </div>
  );
}

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  place: PropTypes.string.isRequired,
  value: PropTypes.string,
  text: PropTypes.string,
  error: PropTypes.string,
  handleChange: PropTypes.func.isRequired
};

export default React.memo(TextInput);
