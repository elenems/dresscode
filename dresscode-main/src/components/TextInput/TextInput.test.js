import { shallow } from "enzyme";
import React from "react";
import TextInput from "./TextInput";

describe("TextInput", () => {
  const props = {
    name: "title",
    place: "title",
    value: "",
    handleChange: () => {},
    type: "text",
    error: "",
    required: true
  };
  it("Matches snapshot", () => {
    const wrapper = shallow(<TextInput {...props} />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Shows error on provided error", () => {
    props.error = "Error";
    const wrapper = shallow(<TextInput {...props} />);
    expect(wrapper.find(".error").text()).toBe("Error");
  });

  it("Shows label if required", () => {
    props.required = false;
    const wrapper = shallow(<TextInput {...props} />);
    expect(
      wrapper.contains(
        <label className="required" htmlFor={props.name}>
          {props.name}
        </label>
      )
    ).toBe(true);
  });
});
