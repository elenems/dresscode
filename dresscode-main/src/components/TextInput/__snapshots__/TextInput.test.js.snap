// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Matches snapshot 1`] = `
"<div className=\\"input-container\\">
  <label htmlFor=\\"title\\">
    title
  </label>
  <input style={{...}} onChange={[Function: handleChange]} id=\\"title\\" type=\\"text\\" value=\\"\\" placeholder=\\"title\\" />
</div>"
`;
